test/unit/mem/buf/src/ManagedAndMappedMemoryBufTest.cpp

```diff
@@ -0,0 +1,145 @@
+    auto const dev = alpaka::getDevByIdx(platformAcc, 0);
+
+    INFO("Test if unified memory works in: ");
+    INFO(alpaka::getName(dev));
+
+    Queue queue(dev);
+
+    // Configure a 1D work division large enough to cover all elements
+    Idx const numElements(123456);
+    Idx const elementsPerThread(1u);
+
+    alpaka::Vec<Dim, Idx> extent = alpaka::Vec<Dim, Idx>::ones();
+    extent[0] = numElements;
+
+    alpaka::Vec<Dim, Idx> elemsPerThreadVec = alpaka::Vec<Dim, Idx>::ones();
+    elemsPerThreadVec[0] = elementsPerThread;
+
+    // Allocate unified memory associated with the host device, accessible by both host and accelerator
+    auto buf = alpaka::allocManagedBuf<Elem, Idx>(devHost, platformAcc, extent);
+
+    constexpr Elem fillVal = 42;
+    auto* hostPtr = alpaka::getPtrNative(buf);
+    for(Idx i = 0; i < numElements; ++i)
+    {
+        hostPtr[i] = fillVal;
+    }
+
+    constexpr int value = 10;
+
+    ValueAddKernel kernel;
+    alpaka::KernelCfg<Acc> const kernelCfg = {extent, elemsPerThreadVec};
+
+    // Calculate work division
+    auto const workDiv = alpaka::getValidWorkDiv(kernelCfg, dev, kernel, buf.data(), value, numElements);
+
+    std::cout << "Testing Kernel with scalar indices with a grid of "
+              << alpaka::getWorkDiv<alpaka::Grid, alpaka::Blocks>(workDiv) << " blocks x "
+              << alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(workDiv) << " threads x "
+              << alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(workDiv) << " elements...\n";
+
+    // Call of kernel to change the values of the buffer's elements
+    auto const taskKernel = alpaka::createTaskKernel<Acc>(workDiv, kernel, buf.data(), value, numElements);
+
+    alpaka::enqueue(queue, taskKernel);
+    alpaka::wait(queue);
+
+    // Verify that device writes are immediately visible on host without explicit copy
+    bool passed = true;
+    for(Idx i = 0; i < numElements; ++i)
+    {
+        if(hostPtr[i] != fillVal + value)
+        {
+            passed = false;
+            break;
+        }
+    }
+    CHECK(passed);
+}
+
+TEMPLATE_LIST_TEST_CASE("memBufMappedTest", "[memBuf]", alpaka::test::TestAccs)
+{
+    using Acc = TestType;
+    using Dev = alpaka::Dev<Acc>;
+    using Queue = alpaka::test::DefaultQueue<Dev>;
+    using Elem = int;
+    using Dim = alpaka::Dim<Acc>;
+    using Idx = alpaka::Idx<Acc>;
+
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+
+    auto const platformAcc = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platformAcc, 0);
+
+    INFO("Test if mapped memory works in: ");
+    INFO(alpaka::getName(dev));
+
+    Queue queue(dev);
+
+    auto const extent = alpaka::test::extentBuf<Dim, Idx>;
+
+    auto buf = alpaka::allocMappedBuf<Elem, Idx>(devHost, platformAcc, extent);
+
+    constexpr Elem fillVal = 42;
+    alpaka::fill(queue, buf, fillVal);
```

Comment:
IMO this should be an explicit kernel because for SYCL the fill will be mapped to the native fill method and it is not necessary a kernel that is executed on the device.